### ISSUE TITLE
WebJobs not deployed by Kudu due to asp.net 5 trickery or some other business

### ISSUE THREAD

**Issue by tonysurma** [HTBox/allReady#348]
Simply put the Kudu deployment does not deploy webjobs.  I currently manually deploy to -mvp and it works but this is not sustainable as I tend to sleep and do other things from time to time :smile:

**Comment by tonysurma** [comment]
pinging @dpaquette and @mheggeseth as one worked on the Kudu with @BillWagner and I and the other has some history with webjobs

**Comment by MisterJames** [comment]
@tonysurma I have done a deep dive into the world of web jobs. There's actually a lot less magic going on than I assumed! :)

TLDR: the fix is just a file copy to a well-known location.

    wwwroot/app_data/jobs/continuous/web_job_name

First off, I'm not sure what you were doing to manually deploy it (upload via portal, perhaps?), but you were onto the right track. :heavy_check_mark: After deploying a project with a web job to my Azure account, there is virtually no difference to the site, except that the job appears in the "continuous" folder at the above location. 

#The Ugly
Surprise, but it looks like we're hitting a big of a kudu/aspnet5 snag. Without diving in and modifying xproj files (and not being sure what would happen there down the road), I can't figure out how to make proper use of the NuGet package that, in Asp.Net 4 happily made the job deploy. There are a few files that are added and a bit of config - it's actually pretty trivial - but it also modifies the build targets of both the "receiving" web site and the "pushing" web job.  I just have no visibility as to what might come if I start messing with the xproj in our Asp.Net 5 project. 

#The Good
Honestly, it looks like all the config and the package are just set up to take care of the copy operation. It is actually a little more complex if we are working with scheduled jobs, but in continuous it looks like it's just picking up the name and binaries. **We can do that from our kudu script.**

:fireworks: 

So, we'll need to compile the web job (we're already doing this) and copy the output to the prescribed path. My best guess is that we're going to need to copy everything from here:

    "%DEPLOYMENT_SOURCE%\NotificationsProcessor\bin\debug\"

...to here:

    "%DEPLOYMENT_TEMP%\site\wwwroot\app_data\jobs\continuous\notificationsprocessor\"
 
Note that if we're doing a release build instead of a debug build (we're currently not doing a `/p:Configuration=Release` param) then the above path will be `\bin\release` instead.

> **The `xcopy` operation should be done in section `4.1. Publish the site`** so that the files live under the web site path. Because we nest it under the web site path, the files get copied over during the kudu sync step. 

My only hesitation is whether or not we need the `site\wwwroot` components in the target path; this will have to be tested/verified.

@BillWagner I'm going to tag you in here as you already have the bits set up to tackle testing this. If you would like me to walk through it with you, please let me know and we'll jump on Skype or something.

**Comment by tonysurma** [comment]
@MisterJames that is exactly what my quick research showed we could 'hack' to make it work for now.

What do you mean by "My only hesitation is whether or not we need the  site\wwwroot  components in the target path; this will have to be tested/verified." ?  Those files are already created by the publish and there to be synched by kudu so if we copy our items to the app_data/jobs path after the dnx publish and before the kudu synch we should be good.

**Comment by MisterJames** [comment]
Yes, we're on the same page. I just don't know from the current cmd what the relative path is for the kudu script. So, can't confirm if we need site\ in the target path, \wwwroot or both.

I can confirm the order is dnu publish the site, msbuild the web job, copy all files from WJ to WS, then finally let kudu sync take over.

**Comment by tonysurma** [comment]
Gotcha, Yep.  @BillWagner can test that out in his test deployment site.  I think it is deploying to \site so we don't need site but need wwwroot (if memory serves correctly)

**Comment by BillWagner** [comment]
@MisterJames What is WJ and WS? Not familiar with those acronyms . (Or just tired and haven't had coffee yet).

**Comment by MisterJames** [comment]
Web job and web site :coffee: :)

**Comment by MisterJames** [comment]
@BillWagner perhaps later this week we can work on this together. For now, @tonysurma will have to manually deploy the web job updates that we have made.

**Comment by BillWagner** [comment]
@MisterJames It's on my list for today. I'll work on it this afternoon, and ping you if I need help.
